src/audio_pixel.h

```diff
@@ -37,14 +40,50 @@ class audio_pixel_t {
 class audio_pixel_block_t {
 
 public: 
-    // TODO: handle the accessor not being valid
-    audio_pixel_block_t(MediaItem_Take* take, double pix_per_s) 
-        : m_take(take), m_pix_per_s(pix_per_s), 
-          m_accessor(safe_audio_accessor_t(take)) 
-          { /* TODO: */}
+    audio_pixel_block_t(double pix_per_s)
+        :m_pix_per_s(pix_per_s) {}
     ~audio_pixel_block_t() {}
 
-    void update() { /* TODO: see AudioAccessorStateChanged */}
+    void update(std::vector<double>& sample_buffer, int num_channels, int sample_rate) {
+        // constructs an audio pixel block using interleaved sample buffer and num channels
+        // clear the current set of audio pixels 
+        m_pixels.clear();
+
+        // vectors to store each channels audio pixel data
+        std::vector<double> channel_max(num_channels);
```

Comment:
why do we need to store vector this? 

you could create the `audio_pixel_t` from the start. (although you would have to publicize the `m_max`, `m_min`, and `m_rms` members, which I'm ok with). 
initally, assign `m_max = -DBL_MAX`, `m_min = DBL_MIN`, and `m_rms = 0`. 

then, update the max with `std::max(pixel.m_max, cur_sample)`, the min with `std::min(pixel.m_min, cur_sample)`, so on.

Reply:
just out of curiosity, since the `auio_pixel_t` class only has public fields now would you want to make it a struct? i don't think we'll need anything class bound methods for the `audio_pixel_t`

Reply:
yeah let's make it a struct and remove the `m_` prefixes